addons/account_payment_term/models/account_payment_term.py

```diff
@@ -0,0 +1,46 @@
+from dateutil.relativedelta import relativedelta
+from odoo import api, fields, models, _
+from odoo.exceptions import ValidationError
+from odoo.tools import date_utils
+class AccountPaymentTermLine(models.Model):
+    _inherit = "account.payment.term.line"
+
+    delay_type = fields.Selection(
+        selection_add=[('days_end_of_month_on_the', 'Days end of month on the')],
+        ondelete={'days_end_of_month_on_the': 'set default'}
+    )
+    display_days_next_month = fields.Boolean(compute='_compute_display_days_next_month')
+    days_next_month = fields.Char(
```

Comment:
For the reviewer: The fact that i've put a char is to block the size of the field to 2 since size doesn't exit for integer